src/masked-input/types.ts

```diff
@@ -13,6 +13,7 @@ export interface MaskedInputProps extends InputProps {
     onMask?: (() => void) | undefined;
     onUnmask?: (() => void) | undefined;
     disableMask?: boolean | undefined;
+    transformUppercase?: boolean;
```

Comment:
Thinking for more flexibility, let's make it a string literal type
```
transform?: "uppercase" | "lowercase" | undefined;
```

Reply:
m just thinking if we need these transform, cause our nric validator fails any non-uppercase input, should uppercase be the default?

Reply:
I was thinking of the possibility of this component being reused for other purposes such as password field, so one way is to just pass the props later on, depending on the validator behaviour. I think going for flexibility would make more sense now?

Reply:
yup agree. I am seeing this more than just for nric. The user for nric fields should set the prop to `uppercase`